odl/discr/lp_discr.py

```diff
@@ -775,19 +778,47 @@ def show(self, title=None, method='', coords=None, indices=None,
             indices = []
             for axis, (xmin, xmax, n, coord) in enumerate(
                     zip(interv.min_pt, interv.max_pt, shape, coords)):
-                if coord is None:
-                    indices += [slice(None)]
-                else:
-                    if not xmin <= coord <= xmax:
+                try:
+                    coord_minp, coord_maxp = coord
+                except TypeError:
+                    coord_minp = coord_maxp = coord
+
+                # Validate input
+                if coord_minp is not None:
+                    coord_minp = float(coord_minp)
+                    if not xmin <= coord_minp <= xmax:
+                        raise ValueError('in axis {}: coordinate {} not in '
+                                         'the valid range {}'
+                                         ''.format(axis, coord, [xmin, xmax]))
+                if coord_maxp is not None:
+                    coord_maxp = float(coord_maxp)
+                    if not xmin <= coord_maxp <= xmax:
                         raise ValueError('in axis {}: coordinate {} not in '
                                          'the valid range {}'
                                          ''.format(axis, coord, [xmin, xmax]))
 
-                    if xmin == xmax:
-                        indices += [0]
+                if xmin == xmax:  # trivial cases
+                    indices += [0]
+                elif coord_minp is not None and coord_minp == coord_maxp:
+                    normalized_pos = ((float(coord_minp) - xmin) /
```

Comment:
Minor thing, you can remove the cast here and further down also. Merge after that change.